dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -157,9 +165,33 @@ protected void getClientHostDescription() {
     protected String getUserAppJarLocation(String userAppJar) {
         String jarPath = null;
         jarPath = dtestsLibsDir + userAppJar;
+        if (new File(jarPath).exists()) return jarPath;
+        jarPath = productLibsDir + userAppJar;
         if (!new File(jarPath).exists()) {
-            String s = "User App jar doesn't exists at expected location: " + dtestsLibsDir;
-            throw new TestException(s);
+            jarPath = clusterLibsDir + userAppJar;
+            if (new File(jarPath).exists()) return jarPath;
+            else
+                jarPath = storeTestsSqlLibsDir + userAppJar;
+            if (new File(jarPath).exists()) return jarPath;
+            else
+                jarPath = storeTestsCoreLibsDir + userAppJar;
+            if (new File(jarPath).exists()) return jarPath;
+            else
+                jarPath = sparkStreamingLibsDir + userAppJar;
+            if (new File(jarPath).exists()) return jarPath;
+            else
+                jarPath = sparkExamplesLibsDir + userAppJar;
+            if (new File(jarPath).exists()) return jarPath;
+            else
+                jarPath = snappyExamplesLibsDir + userAppJar;
+            if (new File(jarPath).exists()) return jarPath;
+            else
+                jarPath = aqpLibsDir + userAppJar;
+            if (new File(jarPath).exists()) return jarPath;
+            else {
+                String s = "User App jar doesn't exists at any expected location.";
+                throw new TestException(s);
+            }
```

Comment:
you could have  used the File.listFiles(FileNameFilter filter) to  get the jar file  name in the directory instead of searching in individual directory.